drivers/ethernet/eth_nxp_imx_netc.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_ETHERNET_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nxp_imx_eth);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_pkt.h>
+#include <zephyr/net/phy.h>
+#include <ethernet/eth_stats.h>
+
+#include "eth.h"
+#include "eth_nxp_imx_netc_priv.h"
+
+const struct device *netc_dev_list[NETC_DRV_MAX_INST_SUPPORT];
+
+static int netc_eth_rx(const struct device *dev)
+{
+	struct netc_eth_data *data = dev->data;
+	struct net_pkt *pkt;
+	int key;
+	int ret = 0;
+	status_t result;
+	uint32_t length;
+
+	key = irq_lock();
+
+	/* Check rx frame */
+	result = EP_GetRxFrameSize(&data->handle, 0, &length);
+	if (result == kStatus_NETC_RxFrameEmpty) {
+		ret = -ENOBUFS;
+		goto out;
+	}
+
+	if (result != kStatus_Success) {
+		LOG_ERR("Error on received frame");
+		ret = -EIO;
+		goto out;
+	}
+
+	/* Receive frame */
+	result = EP_ReceiveFrameCopy(&data->handle, 0, data->rx_frame, length, NULL);
+	if (result != kStatus_Success) {
+		LOG_ERR("Error on received frame");
+		ret = -EIO;
+		goto out;
+	}
+
+	/* Copy to pkt */
+	pkt = net_pkt_rx_alloc_with_buffer(data->iface, length, AF_UNSPEC, 0, NETC_TIMEOUT);
+	if (!pkt) {
+		eth_stats_update_errors_rx(data->iface);
+		ret = -ENOBUFS;
+		goto out;
+	}
+
+	ret = net_pkt_write(pkt, data->rx_frame, length);
+	if (ret) {
+		eth_stats_update_errors_rx(data->iface);
+		net_pkt_unref(pkt);
+		goto out;
+	}
+
+	/* Send to upper layer */
+	ret = net_recv_data(data->iface, pkt);
+	if (ret < 0) {
+		eth_stats_update_errors_rx(data->iface);
+		net_pkt_unref(pkt);
+		LOG_ERR("Failed to enqueue frame into rx queue: %d", ret);
+	}
+out:
+	irq_unlock(key);
+	return ret;
+}
+
+static void netc_eth_rx_thread(void *arg1, void *unused1, void *unused2)
+{
+	const struct device *dev = (const struct device *)arg1;
+	struct netc_eth_data *data = dev->data;
+	int ret;
+	int work;
+
+	ARG_UNUSED(unused1);
+	ARG_UNUSED(unused2);
+
+	while (1) {
+		ret = k_sem_take(&data->rx_sem, K_FOREVER);
+		__ASSERT_NO_MSG(ret == 0);
```

Comment:
is it really a good idea to put an assert in here? won't it cause a big system failure? can we handle the error?

Reply:
Will handle it here.